.changeset/nine-spoons-matter.md

```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Yield` - Converted component to typescript
```

Comment:
We don't need a changelog entry for this, as it's an internal/utility component and is not part of the public API.